cli.js

```diff
@@ -78,14 +78,19 @@ xfs.mkdirp(program.output, async err => {
 
   // If we want to watch for changes do that
   if (program.watchTemplate) {
-    const watchDir = path.resolve(Generator.DEFAULT_TEMPLATES_DIR, template);
+    const watchDir = path.resolve(template);
+    // Template name is needed as it is not always a part of the cli commad
+    // There is a use case that you run generator from a root of the template with `./` path
+    const templateName = require(path.resolve(watchDir,'./package.json')).name;
```

Comment:
Not sure if the slash there will create problems on Windows. Just in case, you can do this:

```suggestion
    const templateName = require(path.resolve(watchDir,'package.json')).name;
```